pkg/apps/registry/rollback/rollback_generator.go

```diff
@@ -29,18 +30,16 @@ func NewRollbackGenerator() RollbackGenerator {
 
 type rollbackGenerator struct{}
 
-func (g *rollbackGenerator) GenerateRollback(from, to *appsapi.DeploymentConfig, spec *appsapi.DeploymentConfigRollbackSpec) (*appsapi.DeploymentConfig, error) {
-	rollback := &appsapi.DeploymentConfig{}
+func (g *rollbackGenerator) GenerateRollback(from, to *appsv1.DeploymentConfig, spec *appsv1.DeploymentConfigRollbackSpec) (*appsapi.DeploymentConfig, error) {
```

Comment:
shouldn't this be internal as it's deal with only at REST?

Reply:
@tnozicka the helper this require is being used in external and I hate to have duplicated Decode just to support rollback generator (isn't it dead btw?)

Reply:
which helper?

I'd do the convertion around the helper, not at the method args. now you have args of external and return internal

Reply:
to, err := appsutil.DecodeDeploymentConfig(targetDeployment)